### DISCUSSION TITLE
PZEM 004T V3 Impossible to reset hardware counter

### DISCUSSION THREAD

**Discussion by ster1um** [arendst/Tasmota#13291] [Ideas]
### PROBLEM DESCRIPTION
After activation the setoption72 is not possible reset counter on pzem004 v3 via console.
If use command energyreset3 0 the counter on tasmota reset but after reload the pzem total .counter

### REQUESTED INFORMATION
_Make sure your have performed every step and checked the applicable boxes before submitting your issue. Thank you!_

- [ ] Read the [Contributing Guide and Policy](https://github.com/arendst/Tasmota/blob/development/CONTRIBUTING.md) and [the Code of Conduct](https://github.com/arendst/Tasmota/blob/development/CODE_OF_CONDUCT.md)
- [ ] Searched the problem in [issues](https://github.com/arendst/Tasmota/issues)
- [ ] Searched the problem in [discussions](https://github.com/arendst/Tasmota/discussions)
- [ ] Searched the problem in the [docs](https://tasmota.github.io/docs/FAQ)
- [ ] Searched the problem in the [chat](https://discord.gg/Ks2Kzd4)
- [ ] Device used (e.g., Sonoff Basic): wemos d1 mini + pzem003t v3.0
- [ ] Tasmota binary firmware version number used: 9.5 stable
  - [ X] Pre-compiled
  - [ ] Self-compiled
- [ ] Flashing tools used: _____
- [ ] Provide the output of command: `Backlog Template; Module; GPIO 255`:
```lua
  Configuration output here:


```
- [ ] If using rules, provide the output of this command: `Backlog Rule1; Rule2; Rule3`:
```lua
  Rules output here:


```
- [ ] Provide the output of this command: `Status 0`:
```lua
  STATUS 0 output here:
11:42:59.819 CMD: STATUS 0
11:42:59.821 SRC: WebConsole from 192.168.2.101
11:42:59.824 CMD: Grp 0, Cmnd 'STATUS', Idx 1, Len 1, Data '0'
11:42:59.830 MQT: stat/Contatore_Energia/STATUS = {"Status":{"Module":0,"DeviceName":"Contatore Energia","FriendlyName":["Contatore Energia"],"Topic":"Contatore_Energia","ButtonTopic":"0","Power":0,"PowerOnState":3,"LedState":1,"LedMask":"FFFF","SaveData":1,"SaveState":1,"SwitchTopic":"0","SwitchMode":[0,0,0,0,0,0,0,0],"ButtonRetain":0,"SwitchRetain":0,"SensorRetain":0,"PowerRetain":0,"InfoRetain":0,"StateRetain":0}}
11:42:59.837 MQT: stat/Contatore_Energia/STATUS1 = {"StatusPRM":{"Baudrate":9600,"SerialConfig":"8N1","GroupTopic":"tasmotas","OtaUrl":"http://ota.tasmota.com/tasmota/release/tasmota-minimal.bin.gz","RestartReason":"Software/System restart","Uptime":"0T00:21:04","StartupUTC":"2021-10-07T09:21:55","Sleep":50,"CfgHolder":4617,"BootCount":28,"BCResetTime":"2021-10-05T16:18:51","SaveCount":128,"SaveAddress":"F8000"}}
11:42:59.844 MQT: stat/Contatore_Energia/STATUS2 = {"StatusFWR":{"Version":"9.5.0(tasmota)","BuildDateTime":"2021-06-17T08:26:35","Boot":31,"Core":"2_7_4_9","SDK":"2.2.2-dev(38a443e)","CpuFrequency":80,"Hardware":"ESP8266EX","CR":"387/699"}}
11:42:59.850 MQT: stat/Contatore_Energia/STATUS3 = {"StatusLOG":{"SerialLog":0,"WebLog":3,"MqttLog":0,"SysLog":1,"LogHost":"192.168.2.201","LogPort":514,"SSId":["MQTT",""],"TelePeriod":11,"Resolution":"558180C0","SetOption":["00008009","2805C8000100060000005A0A000000000000","00400080","00006000","00000000"]}}
11:42:59.863 MQT: stat/Contatore_Energia/STATUS4 = {"StatusMEM":{"ProgramSize":607,"Free":396,"Heap":25,"ProgramFlashSize":1024,"FlashSize":4096,"FlashChipId":"164020","FlashFrequency":40,"FlashMode":3,"Features":["00000809","8FDAC787","04368001","000000CF","010013C0","C000F981","00004004","00001000","00000020"],"Drivers":"1,2,3,4,5,6,7,8,9,10,12,16,18,19,20,21,22,24,26,27,29,30,35,37,45","Sensors":"1,2,3,4,5,6"}}
11:42:59.871 MQT: stat/Contatore_Energia/STATUS5 = {"StatusNET":{"Hostname":"Contatore_Energia","IPAddress":"192.168.2.138","Gateway":"192.168.2.3","Subnetmask":"255.255.255.0","DNSServer":"192.168.2.3","Mac":"F4:CF:A2:64:E6:49","Webserver":2,"WifiConfig":4,"WifiPower":17.0}}
11:42:59.877 MQT: stat/Contatore_Energia/STATUS6 = {"StatusMQT":{"MqttHost":"192.168.2.201","MqttPort":1883,"MqttClientMask":"DVES_%06X","MqttClient":"DVES_64E649","MqttUser":"mqtt","MqttCount":1,"MAX_PACKET_SIZE":1200,"KEEPALIVE":30,"SOCKET_TIMEOUT":4}}
11:42:59.886 MQT: stat/Contatore_Energia/STATUS7 = {"StatusTIM":{"UTC":"2021-10-07T09:42:59","Local":"2021-10-07T11:42:59","StartDST":"2021-03-28T02:00:00","EndDST":"2021-10-31T03:00:00","Timezone":99,"Sunrise":"07:59","Sunset":"19:16"}}
11:42:59.892 MQT: stat/Contatore_Energia/STATUS9 = {"StatusPTH":{"PowerDelta":[0,0,0],"PowerLow":0,"PowerHigh":0,"VoltageLow":0,"VoltageHigh":0,"CurrentLow":0,"CurrentHigh":0}}
11:42:59.902 MQT: stat/Contatore_Energia/STATUS10 = {"StatusSNS":{"Time":"2021-10-07T11:42:59","ENERGY":{"TotalStartTime":"2021-10-07T11:37:35","Total":0.000,"Yesterday":0.000,"Today":0.000,"Power":4,"ApparentPower":10,"ReactivePower":9,"Factor":0.39,"Frequency":50,"Voltage":226,"Current":0.045}}}
11:42:59.909 MQT: stat/Contatore_Energia/STATUS11 = {"StatusSTS":{"Time":"2021-10-07T11:42:59","Uptime":"0T00:21:04","UptimeSec":1264,"Heap":25,"SleepMode":"Dynamic","Sleep":50,"LoadAvg":19,"MqttCount":1,"Wifi":{"AP":1,"SSId":"MQTT","BSSId":"06:26:5A:CB:F3:54","Channel":1,"Mode":"11n","RSSI":92,"Signal":-54,"LinkCount":1,"Downtime":"0T00:00:03"}}}


```
- [ ] Set `weblog` to 4 and then, when you experience your issue, provide the output of the Console log:
```lua
  Console output here:
11:44:17.755 CMD: energyreset3 0
11:44:17.757 SRC: WebConsole from 192.168.2.101
11:44:17.759 CMD: Grp 0, Cmnd 'ENERGYRESET', Idx 3, Len 1, Data '0'
11:44:17.766 MQT: stat/Contatore_Energia/RESULT = {"EnergyReset":{"Total":0.000,"Yesterday":0.000,"Today":0.000,"Usage":[0.000,3.061],"Export":[0.000,0.000]}}
11:44:18.102 CFG: Saved to flash at F7, Count 129, Bytes 4096

11:45:43.776 CMD: energyreset
11:45:43.778 SRC: WebConsole from 192.168.2.101
11:45:43.781 CMD: Grp 0, Cmnd 'ENERGYRESET', Idx 1, Len 0, Data ''
11:45:43.788 MQT: stat/Contatore_Energia/RESULT = {"EnergyReset":{"Total":1.521,"Yesterday":0.000,"Today":0.004,"Usage":[0.000,3.061],"Export":[0.000,0.000]}}


```

### TO REPRODUCE
_Steps to reproduce the behavior:_

### EXPECTED BEHAVIOUR
_A clear and concise description of what you expected to happen._

### SCREENSHOTS
_If applicable, add screenshots to help explain your problem._

### ADDITIONAL CONTEXT
_Add any other context about the problem here._

**(Please, remember to close the issue when the problem has been addressed)**

**Comment by Jason2866** [comment]
Please try latest development version.

**Reply by ster1um** [reply]
> 
> 
> Please try latest development version.

This issue is moved to @ascillato2 from https://github.com/arendst/Tasmota/issues/13289

Tasmota fimware don't have this featured!

Hope for a future update.

**Comment by FARKIr** [comment]
13:42:36.117 CMD: Backlog Template; Module; GPIO 255
13:42:36.229 MQT: stat/tasmota_2EC9F8/RESULT = {"NAME":"PZEM address1","GPIO":[32,1472,1,1536,1,1,1,1,224,1,288,1,1,1],"FLAG":0,"BASE":18}
13:42:36.438 MQT: stat/tasmota_2EC9F8/RESULT = {"Module":{"18":"Generic"}}
13:42:36.694 MQT: stat/tasmota_2EC9F8/RESULT = {"GPIO0":{"32":"Tlačidlo1"},"GPIO1":{"1472":"PZEM0XX Tx"},"GPIO2":{"0":"Žiaden"},"GPIO3":{"1536":"PZEM016 Rx"},"GPIO4":{"0":"Žiaden"},"GPIO5":{"0":"Žiaden"},"GPIO9":{"0":"Žiaden"},"GPIO10":{"0":"Žiaden"},"GPIO12":{"224":"Relé1"},"GPIO13":{"0":"Žiaden"},"GPIO14":{"288":"Led1"},"GPIO15":{"0":"Žiaden"},"GPIO16":{"0":"Žiaden"},"GPIO17":{"0":"Žiaden"}}

**Comment by FARKIr** [comment]
13:44:08.578 CMD: Backlog Rule1; Rule2; Rule3
13:44:08.655 MQT: stat/tasmota_2EC9F8/RESULT = {"Rule1":{"State":"OFF","Once":"OFF","StopOnError":"OFF","Length":0,"Free":511,"Rules":""}}
13:44:08.867 MQT: stat/tasmota_2EC9F8/RESULT = {"Rule2":{"State":"OFF","Once":"OFF","StopOnError":"OFF","Length":0,"Free":511,"Rules":""}}
13:44:09.118 MQT: stat/tasmota_2EC9F8/RESULT = {"Rule3":{"State":"OFF","Once":"OFF","StopOnError":"OFF","Length":0,"Free":511,"Rules":""}}

**Comment by FARKIr** [comment]
Controls
PZEM address1
Sensors
ENERGY ApparentPower
0 VA
ENERGY Current
0,0 A
ENERGY Factor
0,0 Cos φ
ENERGY Power
0 W
ENERGY ReactivePower
0 VAr
ENERGY Today
0,0 kWh
ENERGY Total
0,0 kWh
ENERGY TotalStartTime
2022-11-09T12:04:03
ENERGY Voltage
0 V
ENERGY Yesterday
0,0 kWh

**Comment by FARKIr** [comment]
me reading are same 0
 https://prnt.sc/_qPnyKj3s68n

**Reply by barbudor** [reply]
How is your messages related to the original discussion ?
Just posting logs and config and explaining anything won't get you anywhere

**Reply by FARKIr** [reply]
https://prnt.sc/somh7WiZX5pW

https://prnt.sc/SgrC_zEMBcfs

**Reply by FARKIr** [reply]
https://prnt.sc/Xl0tr8919YbD
